### PR TITLE
Support multi-arch builds

### PR DESCRIPTION
**Pull request by howardjohn**
This will be needed to publish the base images as multi-architecture

cc @morlay

### CODE REVIEW

File: tools/build-base-images.sh [hunk 1]
```diff
@@ -19,7 +19,14 @@
 
 set -ex
 
-HUB="${HUB:-istio.io/docker}"
+HUB="${HUB:?specify a hub}"
 TAG="${TAG:?specify a tag}"
 
+# For multi architecture building:
+# See https://medium.com/@artur.klauser/building-multi-architecture-docker-images-with-buildx-27d80f7e2408 for more info
+# * docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```
Comment (morlay) [inline tools/build-base-images.sh:27]: consider use this instead.
```
docker run --privileged --rm tonistiigi/binfmt --install all
```
https://github.com/docker/buildx#building-multi-platform-images

`tonistiigi/binfmt:latest` is a multi arch image with qemu@5.0.0 (multiarch/qemu-user-static only support amd64, and with qemu@4)

Comment (howardjohn) [reply tools/build-base-images.sh:27]: Thanks!

File: tools/buildx-gen.sh [hunk 2]
```diff
@@ -23,6 +23,20 @@ out="${1}"
 config="${out}/docker-bake.hcl"
 shift
 
+function to_platform_list() {
+  image="${1}"
+  platforms="${2}"
+  # Allow list images verified to work with multi architecture
+  # Eventually this should be everything but the testing images (app_sidecar_)
+  if [[ "${image}" == base || "${image}" == distroless ]]; then
```
Comment (morlay) [inline tools/buildx-gen.sh:31]: `app_sidecar_*` is safe to build multi-arch images

`ubuntu` `debian` and `centos` both provider multi-arch image.

https://hub.docker.com/_/ubuntu?tab=tags&name=xenial
https://hub.docker.com/_/ubuntu?tab=tags&name=bionic
https://hub.docker.com/_/ubuntu?tab=tags&name=focal

https://hub.docker.com/_/debian?tab=tags&name=9
https://hub.docker.com/_/debian?tab=tags&name=10

https://hub.docker.com/_/centos?tab=tags&name=7
https://hub.docker.com/_/centos?tab=tags&name=8

If we publish app_sidecar_* too, should resolve https://github.com/istio/istio/pull/27957
cc @BbolroC

Comment (howardjohn) [reply tools/buildx-gen.sh:31]: Hmm yeah I guess the base ones are, but those are only used by a docker image that is amd only. The reason we have app_sidecar_base and app_sidecar is so we don't build the ubuntu source on every PR which is slow. These are just images for testing though so not the most important

Comment (morlay) [reply tools/buildx-gen.sh:31]: Not sure. I just see @BbolroC need this https://github.com/istio/istio/pull/27957#issuecomment-733674551

Comment (BbolroC) [reply tools/buildx-gen.sh:31]: Hello, all. Thanks for the discussion around my matter. In order not to make both of you confused about #27957 , I would like to add something here.

What I meant in the PR is that I would like to exclude those `app_sidecar_*` targets when I run the command `BUILD_WITH_CONTAINER=0 make dockermake docker` on s390x. I did not mean that I need the multi-arch support for those images. 

Of course, I know I can run `DOCKER_TARGETS="docker.pilot docker.proxyv2 docker.app docker.istioctl docker.operator docker.install-cni" BUILD_WITH_CONTAINER=0 make docker` to have the same result. 

I am wondering if the remaining code change in #27957 can be merged. If you guys do not think it is necessary on master, just let me know. I will close the PR.

**Review by ericvn — APPROVED** [review]

**Pull request merged**